src/api.coffee

```diff
@@ -495,6 +495,8 @@ class Database
                     blobptr = allocate result, 'i8', ALLOC_NORMAL
                     sqlite3_result_blob(cx, blobptr,result.length, -1)
                     _free blobptr
+                  else if result instanceof Date
+                    sqlite3_result_text(cx, result.toISOString(), -1, -1)
```

Comment:
Are you sure this is what you wanted ?
This change allows [user-defined fucntions](http://kripken.github.io/sql.js/documentation/class/Database.html#create_function-dynamic) to return a Date. This is great, but wasn't #334 about adding Date support to prepared statement parameters ?

Reply:
Thanks for the tips @lovasoa 
Looks like I was too quick. I changed the build files and my test passed. Didn't realize there were two similar blocks of code.
Now that you have showed me the direction, let me take my time to fix the issues, add tests, add docs and polish the PR